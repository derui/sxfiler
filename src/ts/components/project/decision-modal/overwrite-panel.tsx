import { h } from "preact";

export type Props = {
  selected: boolean;
};

export const Component: preact.FunctionComponent<Props> = ({ selected }) => {
  return (
    <div class="decision-modal__panel" data-testid="decisionModal-overwritePanel" aria-selected={selected}>
      <p class="decision-modal__label" data-testid="decisionModal-overwriteLabel">
        Overwrite
      </p>
    </div>
  );
};
