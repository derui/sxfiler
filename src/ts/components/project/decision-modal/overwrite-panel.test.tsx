import { h } from "preact";
import { render } from "preact-render-to-string";
import { Component } from "./overwrite-panel";

describe("Project", () => {
  describe("Overwrite decision", () => {
    it("render with overwrite decision", () => {
      const tree = render(<Component selected={false} />);

      expect(tree).toMatchSnapshot();
    });

    it("change style when selected", () => {
      const tree = render(<Component selected={true} />);

      expect(tree).toMatchSnapshot();
    });
  });
});
