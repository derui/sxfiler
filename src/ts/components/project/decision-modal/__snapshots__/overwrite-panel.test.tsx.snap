// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Overwrite decision change style when selected 1`] = `"<div class=\\"decision-modal__panel\\" data-testid=\\"decisionModal-overwritePanel\\" aria-selected=\\"true\\"><p class=\\"decision-modal__label\\" data-testid=\\"decisionModal-overwriteLabel\\">Overwrite</p></div>"`;

exports[`Project Overwrite decision render with overwrite decision 1`] = `"<div class=\\"decision-modal__panel\\" data-testid=\\"decisionModal-overwritePanel\\" aria-selected=\\"false\\"><p class=\\"decision-modal__label\\" data-testid=\\"decisionModal-overwriteLabel\\">Overwrite</p></div>"`;
